DEVELOPERS.md

```diff
@@ -1,225 +1,205 @@
 # Developer documentation
 
-## Installation
+This document covers how to get up and running with developing pixelator.
 
-### Using a dedicated conda/mamba environment (recommended)
+## Table of contents
 
-You will need to [install conda](https://docs.anaconda.com/free/anaconda/install/) to use this method.
-We also recommend to install mamba in your base conda environment and use that command instead of conda.
-This will make the installation of the dependencies much faster.
-
-```shell
-conda activate base
-conda install mamba -c conda-forge
-```
+- [Setup the developer environment](#setup-the-developer-environment)
+- [Pixelator developer tools](#pixelator-developer-tools)
+- [Adding new dependencies](#adding-new-dependencies)
+- [Testing pixelator](#testing-pixelator)
+  - [Pixelator unit tests](#pixelator-unit-tests)
+  - [Pixelator integration tests](#pixelator-integration-tests)
+- [Understanding the pixelator plugin system](#understanding-the-pixelator-plugin-system)
 
-All base dependencies are included in the `environment.yml` file provided in the repo root.
-You can now run the following commands to create your pixelator environment:
 
-> __NOTE__ You can replace all occurrences of `conda` in the following section with `mamba`.
+## Setup the developer environment
 
-- clone the repository, change directory and checkout the `dev` branch
-    ```shell
-    git clone git@github.com:PixelgenTechnologies/pixelator.git
-    cd pixelator
-    git checkout dev
-    ```
+We recommend that you develop pixelator in a separate python virtual environment. There are
+many ways to setup a virtual environment, we recommend using conda. If you have a different
+way that you prefer, you can use that as well.
 
-- create the pixelator conda environment
-    ```shell
-    conda env create -f environment.yml
-    ```
+Below we outline the steps you need to take to setup your development environment using conda.
 
-- activate the conda environment.
-    ```shell
-    conda activate pixelator
-    ```
+### 1. Prepare and checkout the pixelator repository for Github
 
-### With conda, using pip
+Pixelator uses `git-lfs` to manage large files. If you do not have `git-lfs` installed, you can install it using the following commands:
```

Comment:
I was referring to the recommendations for developers. If they are going to commit files, what do we want on git-lfs? Any file over X size? Or certain file formats? I think just "Pixelator uses `git-lfs` to manage large files." doesn't give guidance.

Reply:
It lfs works by adding patterns to match e.g. file extensions or entire directories to the `.gitattribute` file. So for 99% of the time you shouldn't have to worry about whether files should go into git-lfs or not, it should be automatically solved for you. If you think it is important that we provide additional guidance around this I would suggest that we add an additional section on in, since for the purpose of getting your development environment up and running I think it is a distracting detail. Does that sound reasonable to you?